File: CS/DevExpressMvcSchedulerEditable/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace DevExpressMvcSchedulerEF.View
{
    public class HomeController : Controller
    {
        //
        // GET: /Home/

public ActionResult Index()
{
    return View(SchedulerDataHelper.DataObject);
}

        public ActionResult SchedulerPartial()
        {
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);
        }

        public ActionResult EditAppointment()
        {
            UpdateAppointment();
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);
        }
        #region #updateappointment
        static void UpdateAppointment()
{
    EFAppointment insertedAppt = SchedulerExtension.GetAppointmentToInsert<EFAppointment>(SchedulerSettingsHelper.CommonSchedulerSettings, 
        SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources());
    SchedulerDataHelper.InsertAppointment(insertedAppt);

    EFAppointment[] updatedAppt = SchedulerExtension.GetAppointmentsToUpdate<EFAppointment>(SchedulerSettingsHelper.CommonSchedulerSettings, 
        SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources());
    foreach (var appt in updatedAppt)
    {
        SchedulerDataHelper.UpdateAppointment(appt);
    }

    EFAppointment[] removedAppt = SchedulerExtension.GetAppointmentsToRemove<EFAppointment>(SchedulerSettingsHelper.CommonSchedulerSettings,
        SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources());
    foreach (var appt in removedAppt)
    {
        SchedulerDataHelper.RemoveAppointment(appt);
    }

}
        #endregion #updateappointment

    }
}

File: VB/DevExpressMvcSchedulerEditable/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace DevExpressMvcSchedulerEF.View
	Public Class HomeController
		Inherits Controller
		'
		' GET: /Home/

Public Function Index() As ActionResult
	Return View(SchedulerDataHelper.DataObject)
End Function

		Public Function SchedulerPartial() As ActionResult
			Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
		End Function

		Public Function EditAppointment() As ActionResult
			UpdateAppointment()
			Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
		End Function
		#Region "#updateappointment"
		Private Shared Sub UpdateAppointment()
	Dim insertedAppt As EFAppointment = SchedulerExtension.GetAppointmentToInsert(Of EFAppointment)(SchedulerSettingsHelper.CommonSchedulerSettings, SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources())
	SchedulerDataHelper.InsertAppointment(insertedAppt)

	Dim updatedAppt() As EFAppointment = SchedulerExtension.GetAppointmentsToUpdate(Of EFAppointment)(SchedulerSettingsHelper.CommonSchedulerSettings, SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources())
	For Each appt In updatedAppt
		SchedulerDataHelper.UpdateAppointment(appt)
	Next appt

	Dim removedAppt() As EFAppointment = SchedulerExtension.GetAppointmentsToRemove(Of EFAppointment)(SchedulerSettingsHelper.CommonSchedulerSettings, SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources())
	For Each appt In removedAppt
		SchedulerDataHelper.RemoveAppointment(appt)
	Next appt

		End Sub
		#End Region ' #updateappointment

	End Class
End Namespace
